### PR TITLE
Console Encoding code Cleanup

### PR DESCRIPTION
**Pull request by tarekgh**
The changes here are just a clean up for previous changes introduced the Console OS Encoding

### CODE REVIEW

**Comment by tarekgh** [comment]
https://github.com/dotnet/corefx/pull/14649

**Comment by tarekgh** [comment]
@stephentoub this is the clean up changes according to your comments in the linked PR. could you please have a look? Thanks.

CC @jkotas @danmosemsft

File: src/Common/src/System/Text/DBCSDecoder.cs [hunk 1]
```diff
@@ -103,7 +104,7 @@ public unsafe override int GetCharCount(byte* bytes, int count, bool flush)
             if (count < 0)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
 
-            bool excludeLastByte = count > 0 || (!flush && IsLastByteALeadByte(bytes, count));
+            bool excludeLastByte = count > 0 && !flush && IsLastByteALeadByte(bytes, count);
```
Comment (stephentoub) [inline src/Common/src/System/Text/DBCSDecoder.cs]: Is there a test we can add to exercise this? (I'm assuming it's not being exercised as the logic here is changing.)

Comment (tarekgh) [inline src/Common/src/System/Text/DBCSDecoder.cs]: will try to think how to add a test for such case. the issue is we need to set the codepage on the console before requesting any encoding functionality . so we'll need to write some test as the one written in System.Diagnostics.Process

Comment (tarekgh) [inline src/Common/src/System/Text/DBCSDecoder.cs]: by the way, before fixing the condition the code should be working too but it was always handling last byte as lead byte which will not change the behavior but would slow a little the perf.

Comment (stephentoub) [inline src/Common/src/System/Text/DBCSDecoder.cs]: Ok, thanks.  No need to block on this.

**Pull request merged**